### PR TITLE
Products view and single product view

### PR DESCRIPTION
**Pull request by ewelkaw**

### CODE REVIEW

File: ewardrobe/ewardrobe_app/models.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ class Retailer(models.Model):
     name = models.CharField(max_length=100)
 
     def __str__(self):
-        return self.retailer
+        return self.name
```
Comment (esse) [inline ewardrobe/ewardrobe_app/models.py]: 👍

File: ewardrobe/ewardrobe_app/templates/products.html [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        <tbody>
+            {% for product in page_obj %}
+            <tr>
+                <td>{{ product.name }}</td>
+                <td>{{ product.price }}$ </td>
+                <td>{{ product.brand }}</td>
+                <td>{{ product.category }}</td>
+                <td>@{{ product.color }}</td>
+                <td>
+                    <a class="btn btn-primary btn-dark" href="/products/{{product.id}}" name='product' role="button">
```
Comment (esse) [inline ewardrobe/ewardrobe_app/templates/products.html]: why not `url()`?

File: ewardrobe/ewardrobe_app/templates/product.html [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                </div>
+                <div class="form-group">
+                    <label for="amount">Amount</label>
+                    <input class="form-control" type="number" min="0" step="1" data-bind="value:replyNumber" id="amount"
+                        name="amount">
+                </div>
+                <input type="submit" class="btn btn-primary btn-dark mb-3">
+            </form>
+        </div>
+        <img src="http://placekitten.com/500/500" class="mr-3 mt-3">
```
Comment (esse) [inline ewardrobe/ewardrobe_app/templates/product.html]: 🐈

File: ewardrobe/ewardrobe_app/views/product.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+
+class ProductView(View):
+    template_name = "product.html"
+    failure_url = "main"
+
+    def get(self, request, *args, **kwargs):
+        if request.user.is_authenticated:
+            id = kwargs["id"]
+            product = Product.objects.filter(id=id).first()
```
Comment (esse) [inline ewardrobe/ewardrobe_app/views/product.py]: I think `Product.objects.get(id=id)` would work better here, or even:

```suggestion
            product = Product.objects.get(id=kwargs["id"])
```

File: ewardrobe/ewardrobe_app/views/products.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    failure_url = "main"
+    paginate_by = 20
+
+    def get(self, request, *args, **kwargs):
+        if request.user.is_authenticated:
+            products = Product.objects.all()
+            paginator = Paginator(
+                products, self.paginate_by
+            )  # Show 20 products per page.
+            page_number = request.GET.get("page") if request.GET.get("page") else 1
```
Comment (esse) [inline ewardrobe/ewardrobe_app/views/products.py]: how about:

```suggestion
            page_number = request.GET.get("page", 1)
```

File: ewardrobe/ewardrobe_app/templates/products.html (hunk 2 above)
Comment (ewelkaw) [reply ewardrobe/ewardrobe_app/templates/products.html]: done

**Pull request merged**